### DISCUSSION TITLE
hdf5 data format

### DISCUSSION THREAD

**Discussion by MedericQUT** [thliebig/openEMS-Project#254] [Q&A Usage] [answered]
Hi All,
Is anybody know why hdf5 data format has 4 dimension?
EX:

If a read a hdf5 file like:
`[field mesh_h5] = ReadHDF5Dump([Sim_Path '/E_dump_f.h5']);`

Then field has 4 dimension x,y,z,K.

I know x,y,z correspond to the dimensions of the dump box.

But what is the 4th dimension K? why is the size of K always 3?

Kind regards
Mederic

**Comment by LubomirJagos42** [comment, marked as answer]
Hi, this is just random simulation which I set to log .h5 file at frequency domain and open it in Octave:
```matlab
[portEtField portEtMesh] = ReadHDF5Dump(['simulation_output' '/electric field HDF5_electric field port.h5']);
```

K is E-field complex value Ex, Ey, Ez
    E_complex = sqrt(Ex^2 + Ey^2 + Ez^2)
    E_magnitude = abs(E_complex)

https://docs.octave.org/v4.4.0/Complex-Arithmetic.html

![image](https://github.com/user-attachments/assets/2d2abc07-3386-4100-9e34-82bf50a41743)

**Reply by MedericQUT** [reply]
Of course, it is the decomponsition of the field along x,y,z... i feel stupid to have asked.
Thank you @LubomirJagos42  :)

**Reply by LubomirJagos42** [reply]
no problem, I also have 90% of time error somewhere in basics, that's engineering :D
